File: src/primus/pawel/Exercise_031.java
package primus.pawel;

//SquareDigit
//7kyu
//https://www.codewars.com/kata/546e2562b03326a88e000020/train/java
//2021-03-22 T:18:19:20
class Exercise_031 {

    public static int squareDigits(int n) {
        String number = String.valueOf(n);
        StringBuilder resultSB = new StringBuilder();
        for (int i = 0; i < number.length(); i++) {
            resultSB.append(Character.getNumericValue(number.charAt(i)) * Character.getNumericValue(number.charAt(i)));
        }
        return Integer.parseInt(resultSB.toString());
    }

    public static void main(String[] args) {
        squareDigits(9119);

    }
}

File: src/JS/Exercise_031.js

// Gravity Flip
// 8kyu
// https://www.codewars.com/kata/5f70c883e10f9e0001c89673/train/javascript
// 2022-04-24 T:18:58:06

const flip = (d, a) => {
    console.log(a)
    return d === 'R' ?
        Array.from(a).sort((a, b) => a - b) :
        Array.from(a).sort((a, b) => a - b).reverse();
}

console.log(flip('L', [ 13, 2, 4, 7, 93 ]));